src/Compilers/CSharp/Portable/FlowAnalysis/ExitPointsWalker.cs

```diff
@@ -41,27 +37,28 @@ protected override void Free()
             base.Free();
         }
 
-        internal static IEnumerable<StatementSyntax> Analyze(CSharpCompilation compilation, Symbol member, BoundNode node, BoundNode firstInRegion, BoundNode lastInRegion)
+        internal static ImmutableArray<SyntaxNode> Analyze(CSharpCompilation compilation, Symbol member, BoundNode node, BoundNode firstInRegion, BoundNode lastInRegion)
```

Comment:
>SyntaxNode [](http://example.com/codeflow?start=39&length=10)

Is the change from StatementSyntax to SyntaxNode done to support the chage ControlFlowAnalysis.cs? I think we can still return ```ImmutableArray<SyntaxNode>``` here. There is a ```ImmutableArray<T>.CastUp``` API that allows reusing the same underlying array during the cast.